pkg/utils/managedresources/registry.go

```diff
@@ -43,14 +45,26 @@ type object struct {
 // NewRegistry returns a new registry for resources. The given scheme, codec, and serializer must know all the resource
 // types that will later be added to the registry.
 func NewRegistry(scheme *runtime.Scheme, codec serializer.CodecFactory, serializer *json.Serializer) *Registry {
-	var groupVersions []schema.GroupVersion
+	var groupVersions schema.GroupVersions
 	for k := range scheme.AllKnownTypes() {
 		groupVersions = append(groupVersions, k.GroupVersion())
 	}
 
+	// Use set to remove duplicates
+	groupVersions = sets.New(groupVersions...).UnsortedList()
+
+	// Sort groupVersions to ensure groupVersions.Identifier() is stable key
+	// for the map in https://github.com/kubernetes/apimachinery/blob/v0.26.1/pkg/runtime/serializer/versioning/versioning.go#L94
+	sort.Slice(groupVersions, func(i, j int) bool {
+		if groupVersions[i].Group == groupVersions[j].Group {
+			return groupVersions[i].Version == groupVersions[j].Version
```

Comment:
Here we are swapping variables if the version is equal. This is not required.
```suggestion
			return false
```

Reply:
Good catch. 
It actually should be 

```suggestion 
			return groupVersions[i].Version < groupVersions[j].Version
```